File: Week 1/Problem Set 1/credit.c
#include <cs50.h>
#include <stdio.h>

string validation(long card_number);

int main(void)
{
    string verify = validation(get_long("Number: "));
    printf("%s\n", verify);
}

string validation(long card_number)
{
    // Default Values
    int two_num = 0;
    int sum1 = 0;
    int sum2 = 0;
    int i = 0;
    int lenght = 0;

    while (card_number != 0)
    {
        // Adds one to card lenght
        lenght += 1;

        // Saves two first numbers to check card type later
        if (card_number < 100 && card_number > 10)
        {
            two_num = card_number;
        }

        // Adds numbers (once to sum1, then to sum2, and so on)
        if (i == 0)
        {
            sum1 += card_number % 10; // Adds last digit from card number
            i = 1;
        }
        else
        {
            int n = card_number % 10 * 2; // Last digit from card number is multiplied by two
            if (n >= 10) // If the multiplied number has two digits - splits it
            {
                n = (n % 10) + ((n - (n % 10)) / 10); // Adds two digits
            }
            sum2 += n;
            i = 0;
        }

        card_number /= 10; // Removes last digit
    }

    string valid = "INVALID";

    // If the card is valid, then it checks the brand - lenght and first two numbers (only first number if VISA)
    if ((sum1 + sum2) % 10 == 0)
    {
        if ((two_num >= 51 && two_num <= 55) && (lenght == 16))
        {
            valid = "MASTERCARD";
        }
        else if ((two_num == 34 || two_num == 37) && (lenght == 15))
        {
            valid = "AMEX";
        }
        else if ((((two_num - (two_num % 10)) / 10) == 4) && (lenght == 13 || lenght == 16))
        {
            valid = "VISA";
        }
    }

    return valid;
}

File: Week 6 - Python/Problem Set 6/credit.py
from cs50 import get_string

# Numbers lists
a = 0
b = 0

# Get user's card number
number = get_string("Number: ")

# Some values
reverse = number[::-1]
length = len(number)

# Loop through number and add numbers once to a once to b
for i in range(length):
    n = int(reverse[i])

    if i % 2 == 0:
        a += n
    else:
        n *= 2

        # If "n" is two digits long, split it and add
        if n >= 10:
            n = (n % 10) + (n // 10)

        b += n

total = (a + b) % 10

# If last digit in total isn't 0, card is invalid
if total != 0:
    print("INVALID")
    exit()

# Check card brand
if int(number[:2]) in (51, 52, 53, 54, 55) and length == 16:
    print("MASTERCARD")
elif int(number[:2]) in (34, 37) and length == 15:
    print("AMEX")
elif int(number[0]) == 4 and length in (13, 16):
    print("VISA")